Server/controllers/authController.js

```diff
@@ -261,6 +262,88 @@ const resetPasswordController = async (req, res, next) => {
     next(error);
   }
 };
+
+/**
+ * Deletes a user and all associated monitors, checks, and alerts.
+ *
+ * @param {Object} req - The request object.
+ * @param {Object} res - The response object.
+ * @param {Function} next - The next middleware function.
+ * @returns {Object} The response object with success status and message.
+ * @throws {Error} If user validation fails or user is not found in the database.
+ */
+const deleteUserController = async (req, res, next) => {
+  try {
+    const token = req.headers.authorization.split(' ')[1];
+    const decodedToken = jwt.decode(token)
```

Comment:
I was checking the docs and `jwt.decode` actually doesn't verify, so we need to change this to `jwt.verify`, otherwise a user can just create their own token and delete whatever they want with it